atlasdb-config/src/test/java/com/palantir/atlasdb/http/AtlasDbHttpClientsTest.java

```diff
@@ -172,6 +175,41 @@ public void canLiveReloadServersList() {
         unavailableServer.verify(getRequestedFor(urlMatching(TEST_ENDPOINT)));
     }
 
+    @Test
+    public void httpProxyThrowsServiceNotAvailableExceptionIfConfiguredWithZeroNodes() {
+        TestResource testResource = AtlasDbHttpClients.createLiveReloadingProxyWithQuickFailoverForTesting(
+                () -> ImmutableServerListConfig.builder().build(),
+                SslSocketFactories::createSslSocketFactory,
+                proxyConfiguration -> ProxySelector.getDefault(),
+                TestResource.class,
+                UserAgents.DEFAULT_VALUE);
+
+        assertThatThrownBy(testResource::getTestNumber).isInstanceOf(ServiceNotAvailableException.class);
+    }
+
+    @Test
+    public void httpProxyCanBeCommissionedAndDecommissionedIfNodesBecomeUnavailable() {
```

Comment:
`IfNodeAvailabilityChanges`?

Reply:
Yep, makes sense - the test does include the case where the nodes become available as well, actually, so this is better.